code/modules/antagonists/space_dragon/carp_rift.dm

```diff
@@ -19,7 +19,7 @@
 		return
 	var/area/rift_location = get_area(owner)
 	if(!(rift_location in dragon.chosen_rift_areas))
-		owner.balloon_alert(owner, "can't summon a rift here!")
+		owner.balloon_alert(owner, "can't summon a rift here! check your objectives!")
```

Comment:
```suggestion
		owner.balloon_alert(owner, "[rift_location] is not a valid rift location!")
```

I mentioned something like this before. It makes the reason for failure very clear that you are in the wrong area, as opposed to any number of other possible things that could be blocking the ability, and provides data to check against your objectives.

Is an issue that it would provide too much information by letting dragon "ping" any area it is in to tell where it is?